service/worker/deletenamespace/reclaimresources/workflow.go

```diff
@@ -0,0 +1,170 @@
+	}
+
+	// Step 2. Delete namespace.
+	ctx5 := workflow.WithLocalActivityOptions(ctx, localActivityOptions)
+	err = workflow.ExecuteLocalActivity(ctx5, a.DeleteNamespaceActivity, params.Namespace, params.NamespaceID).Get(ctx, nil)
+	if err != nil {
+		return result, fmt.Errorf("%w: DeleteNamespaceActivity: %v", errors.ErrUnableToExecuteActivity, err)
+	}
+
+	logger.Info("Child workflow finished successfully.", tag.WorkflowType(WorkflowName))
+	return result, nil
+}
+
+func deleteWorkflowExecutions(ctx workflow.Context, params ReclaimResourcesParams, a *Activities, logger log.Logger) (ReclaimResourcesResult, error) {
+	deleteAttempt := int32(1)
+	executionsExist := true
+	var result ReclaimResourcesResult
+	for executionsExist {
+		ctx1 := workflow.WithChildOptions(ctx, deleteExecutionsWorkflowOptions)
+		var der deleteexecutions.DeleteExecutionsResult
+		err := workflow.ExecuteChildWorkflow(ctx1, deleteexecutions.DeleteExecutionsWorkflow, params.DeleteExecutionsParams).Get(ctx, &der)
+
+		if err != nil {
+			logger.Error("Unable to execute child workflow.", tag.WorkflowType(deleteexecutions.WorkflowName), tag.Error(err))
+			return result, fmt.Errorf("%w: %s: %v", errors.ErrUnableToExecuteChildWorkflow, deleteexecutions.WorkflowName, err)
+		}
+		result.SuccessCount += der.SuccessCount
+		result.ErrorCount += der.ErrorCount
+
+		for checkAttempt := int32(1); checkAttempt <= checkMaxAttempts; checkAttempt++ {
+			err = workflow.Sleep(ctx, visibilityDelay(der.SuccessCount))
+			if err != nil {
+				return result, err
+			}
+
+			ctx2 := workflow.WithLocalActivityOptions(ctx, localActivityOptions)
+			err = workflow.ExecuteLocalActivity(ctx2, a.CheckExecutionsExistActivity, params.NamespaceID, params.Namespace).Get(ctx, &executionsExist)
+			if err != nil {
+				return result, fmt.Errorf("%w: CheckExecutionsExistActivity: %v", errors.ErrUnableToExecuteActivity, err)
+			}
+
+			if !executionsExist {
+				break
+			}
+			logger.Info("Workflow executions are still not deleted.", tag.WorkflowNamespace(params.Namespace.String()), tag.Attempt(checkAttempt))
+		}
+		if executionsExist {
+			logger.Info("Unable to delete workflow executions. Will try again.", tag.WorkflowNamespace(params.Namespace.String()), tag.Counter(der.ErrorCount), tag.Attempt(deleteAttempt))
+			deleteAttempt++
+		}
+	}
+
+	logger.Info("All workflow executions has been deleted successfully.", tag.WorkflowNamespace(params.Namespace.String()), tag.DeletedExecutionsCount(result.SuccessCount), tag.DeletedExecutionsErrorCount(result.ErrorCount))
+	return result, nil
+}
+
+// visibilityDelay returns approximate delay for workflow to sleep and wait for internal tasks to be processed.
+func visibilityDelay(successCount int) time.Duration {
+	const delayPerWorkflowExecution = 10 * time.Millisecond
+	vd := time.Duration(successCount) * delayPerWorkflowExecution
+	if vd < 1*time.Second {
+		vd = 1 * time.Second
+	}
+	if vd > 10*time.Second {
+		vd = 10 * time.Second
+	}
+	return vd
+}
```

Comment:
why not just wait for 10s always, no need for this small optimization that makes code complex.